sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedMissingDataPointFillSettings.cs

```diff
@@ -19,7 +19,9 @@ public DataFeedMissingDataPointFillSettings()
         internal DataFeedMissingDataPointFillSettings(DataFeedDetail dataFeedDetail)
         {
             FillType = dataFeedDetail.FillMissingPointType;
-            CustomFillValue = dataFeedDetail.FillMissingPointValue;
+            CustomFillValue = dataFeedDetail.FillMissingPointType == DataFeedMissingDataPointFillType.CustomValue
+                ? dataFeedDetail.FillMissingPointValue
+                : null;
```

Comment:
This was the source of the issue in `UpdateDataFeed`. `DataFeedMissingDataPointFillSettings` has two properties: `FillMissingPointType` (the strategy used to fill missing points in time series: NoFilling, PreviousValue, CustomValue, ...). There's another property that must be specified when `CustomValue` is set: `CustomFillValue` (the const value used to fill the missing point).

When the fill type is NOT `CustomValue`, I'd expect the service to not return the custom fill value (or return null). However, it returns `0.0`. When we call `Get` followed by `Update`, we store the 0 returned by the service and try to send it back (which is not allowed, because type is not `CustomValue`). We're doing this workaround, but it may be something to fix on the service side.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/16701